lib/al/Library/Area/AreaObjGroup.cpp

```diff
@@ -0,0 +1,64 @@
+#include "al/Library/Area/AreaObjGroup.h"
+
+#include <basis/seadNew.h>
+
+namespace al {
+AreaObjGroup::AreaObjGroup(const char* groupName, const int capacity) {
+    mGroupName = groupName;
+    mBuffer = nullptr;
+    mCount = 0;
+    mCapacity = capacity;
+}
+
+void AreaObjGroup::createBuffer() {
+    if (mCapacity < 1) {
+        return;
+    }
+    mBuffer = new AreaObj*[mCapacity];
+}
+
+void AreaObjGroup::createBuffer(s32 capacity) {
+    mCapacity = capacity;
+    if (mCapacity < 1) {
+        return;
+    }
+    mBuffer = new AreaObj*[mCapacity];
+}
+
+AreaObj* AreaObjGroup::getAreaObj(const s32 index) const {
+    return mBuffer[index];
+}
+
+AreaObj* AreaObjGroup::getInVolumeAreaObj(const sead::Vector3f& position) const {
+    al::AreaObj *areaObj, *toReturn, *curAreaObj;
+    if (mCount < 1) {
+        return nullptr;
+    }
+
+    curAreaObj = nullptr;
+    for (s32 i = 0; i < mCount; i++) {
+        areaObj = mBuffer[i];
+        if ((curAreaObj == nullptr) || (toReturn = curAreaObj, curAreaObj->getPriority() <= areaObj->getPriority())) {
```

Comment:
Assigning to variables shouldn't happen inside `if` conditions. This logic could be simplified to prevent that.